### PR TITLE
doc : Improve our CONTRIBUTING guide from a newcomer perspective

### PR DESCRIPTION
**Pull request by rohanKanojia**
Updated contributing guide so that it's more organized and helpful for newcomers:
- Added table of contents
- Added information about good first issues and first timers only issues
- Added information about how to build project and use local build of plugins (both maven and gradle)
- Added information about how to debug project
- Added information about how to generate project documentation

These changes have been mostly taken from  [Quarkus CONTRIBUTING.md](https://github.com/quarkusio/quarkus/blob/main/CONTRIBUTING.md) and [LibreOffice Easy Hacks](https://wiki.documentfoundation.org/Development/EasyHacks)

### CODE REVIEW

**Comment by manusa** [comment]
This is OK, but it's not tackling the initial problem.

We need to clearly define the types of good-first-issues.

For example: 

first-timers-only:
These issues are aimed for first-time contributors to get acquainted with the project and the Eclipse Foundation contributing policies.
Users are only allowed to take one of these issues and are also encouraged to tackle one of these before doing more complex code contributions.
These issues are extremely simple to solve. In general, the code changes are already described in the issue.
The main purpose is to ensure that contributors have signed an ECA and are familiar with the GitHub pull request procedures.

good-first-issue:
These issues are aimed for users who have already tackled a first-timers-only and would like to continue their contributing journey on the project.
These issues are slightly more complicated and will require actual work from the contributor.
The main purpose is for contributors to gain confidence while working on the code-base and to keep them engaged in the project.

help-wanted:
These issues are more complex and are aimed for regular contributors of the project or people who are familiar with the code-base.

**Comment by rohanKanojia** [comment]
Shall I add abovementioned paragraphs in `Finding an issue to work on` section?

File: src/content/contributing/index.mdx [hunk 1]
```diff
@@ -18,14 +18,49 @@ Thanks for being interested in contributing to Eclipse JKube!
 
 This includes bug reports, fixes, documentation, examples... But first, please read this page.
 
+## Table of Contents
+- ⚖️ [Legal](#legal)
+- 🐞[Reporting an issue](#reporting-an-issue)
```
Comment (manusa) [inline src/content/contributing/index.mdx:23]: a space seems to be missing here

**Comment by manusa** [comment]
> This is OK, but it's not tackling the initial problem.
> 
> We need to clearly define the types of good-first-issues.
> 
> For example:
> 
> first-timers-only: These issues are aimed for first-time contributors to get acquainted with the project and the Eclipse Foundation contributing policies. Users are only allowed to take one of these issues and are also encouraged to tackle one of these before doing more complex code contributions. These issues are extremely simple to solve. In general, the code changes are already described in the issue. The main purpose is to ensure that contributors have signed an ECA and are familiar with the GitHub pull request procedures.
> 
> good-first-issue: These issues are aimed for users who have already tackled a first-timers-only and would like to continue their contributing journey on the project. These issues are slightly more complicated and will require actual work from the contributor. The main purpose is for contributors to gain confidence while working on the code-base and to keep them engaged in the project.
> 
> help-wanted: These issues are more complex and are aimed for regular contributors of the project or people who are familiar with the code-base.

It seems you did add these sections but I missed them.

Please, merge anything that's not already stated in yours and add the third entry.

I'm not sure if this is too far down in the page and users might miss it.

**Comment by rohanKanojia** [comment]
Shall we split this page into multiple small pages?

File: src/content/contributing/index.mdx [hunk 2]
```diff
@@ -50,18 +90,133 @@ All commits must be [signed-off](https://git-scm.com/docs/git-commit#Documentati
...
+Before contributing to any Open Source project, it's important to have basic understanding of what the project is about.
+It is advised to try out the project as an end user. This project is a collection of Maven and Gradle plugins for [Kubernetes](https://kubernetes.io/)
+and [Red Hat OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift), it's advised to learn about these
+technologies if you're interested in seriously contributing to the project.
+
+You can read some blogs about project or watch videos on [Eclipse JKube YouTube channel](https://www.youtube.com/@jkube) .
+
+#### Finding an issue to work on
+
+You're free to start with any type of contribution, whether it is documentation or some issue you faced while
```
Comment (manusa) [inline src/content/contributing/index.mdx:103]: I would add an `but we advise to start with a first-timers-only issue` somewhere here.

**Comment by manusa** [comment]
> Shall we split this page into multiple small pages?

I don't think that's a good idea.

File: src/content/contributing/index.mdx (hunk 2 above)
Comment (rohanKanojia) [reply src/content/contributing/index.mdx:104]: I have added this sentence:

Screenshot_20240517_092134

**Review by manusa — APPROVED** [review]
I would ditch the emojis, I think the rendering won't be uniform and the text to the right won't be correctly aligned.

Other than that, LGTM, and ready to go :rocket:

**Review by manusa — APPROVED** [review]

**Pull request merged**